### PR TITLE
Feature 14 resolves #14

### LINKED ISSUE #14: investigate possibility for custom naming of dbset, classes and properties
_From @ErikEJ on February 22, 2018 9:10_

https://github.com/aspnet/EntityFrameworkCore/blob/dev/src/EFCore.Design/Scaffolding/Internal/RelationalScaffoldingModelFactory.cs 

https://github.com/aspnet/EntityFrameworkCore/issues/11056#issuecomment-368256127

- [x] Get PR for ICandidateNamingService accepted
- [ ] Add support for efpt.renaming.json (drop empty, commented out sample in project, in there is none)
- [ ] add class for renames ColumnRenamer: Table, Old, New & TableRenamer: Old, New
- [ ] Override ICandiadteNamingService, and rename based on json file

_Copied from original issue: ErikEJ/SqlCeToolbox#640_

### PR DESCRIPTION
**Pull request by tomzre**

### CODE REVIEW

File: src/GUI/EFCorePowerTools/EFCorePowerTools.csproj [hunk 1]
```diff
@@ -170,6 +170,7 @@
     <Content Include="efpt\2.1.3\efpt.exe.config">
       <IncludeInVSIX>true</IncludeInVSIX>
     </Content>
+    <None Include="efpt.renaming.json" />
```
Comment (ErikEJ) [inline src/GUI/EFCorePowerTools/EFCorePowerTools.csproj]: Did you want to provide this just as a sample?

Comment (tomzre) [reply src/GUI/EFCorePowerTools/EFCorePowerTools.csproj]: Yes. Just only as sample but didn't know where to put it.

Comment (ErikEJ) [reply src/GUI/EFCorePowerTools/EFCorePowerTools.csproj]: Yes, I need to think of a good way to make this feature not too hidden.

Let's get this in and think of a good way to do that later.

**Comment by tomzre** [comment]
Is it ok to be merged into master?

**Comment by ErikEJ** [comment]
Merged, thank you very much!

Need to think about how to unhide this feature vua the GUI

**Comment by tomzre** [comment]
I am glad that I could help. Just please be aware of the InverseProperty for FK names that I mention in the #14  topic. I can submit PR with the changes that I made to it if you have time to review it. But right now without it should work fine.

**Comment by ErikEJ** [comment]
Could you explain the issue - is it reated to the fact that this property does not adhere to the renaming?

**Comment by tomzre** [comment]
In the ICandidateNamingService two methods that I've overwritten are responsible for renaming columns and tables. But there 2 more for the foreign keys which they are using a IForeignKey and I couldnt match them with appropriate table. But yes it is going behaviour that InverseProperty does not adhere the renaming.

**Comment by tomzre** [comment]
This is what I am talking about. [https://github.com/aspnet/EntityFrameworkCore/blob/release/2.2/src/EFCore.Design/Scaffolding/Internal/CandidateNamingService.cs#L43](url). We can use this `foreignKey.PrincipalEntityType.ShortName()` and it seems to work good.

**Comment by ErikEJ** [comment]
>>  I can submit PR with the changes that I made to it if you have time to review it. 

Please do! :smile:

**Pull request merged**